isbg/isbg.py

```diff
@@ -476,10 +476,9 @@ def _do_save_password(self):
     def do_list_imap(self):
         """List the imap boxes."""
         imap_list = self.imap.list()
-        dirlist = str([x.decode() for x in imap_list[1]])
-        # string formatting
-        dirlist = re.sub(r'\(.*?\)| \".\" \"|\"\', \'', " ", dirlist)
-        self.logger.info(dirlist)
+        for x in imap_list[1]:
+            x = re.sub(r'\(.*" (?=[a-zA-Z0-9])', "", x)    # string formatting
```

Comment:
Could you expand this comment to explain what exactly is being done here and why, ideally with a reference to the IMAP spec? It's a pretty obscure regex.

Reply:
`self.imap.list()` returns something like this, depending on what IMAP directories you have:

```
('OK', ['(\\HasChildren) "." INBOX', '(\\HasNoChildren \\Trash) "." INBOX.Trash', '(\\HasNoChildren) "." INBOX.Spam', '(\\HasNoChildren \\Drafts) "." INBOX.Drafts'])
```

That regex deletes everything that starts with `(\` and stops at `" `. The goal is to delete the `(\\HasChildren) "." ` or `(\\HasNoChildren \\FOO) "." ` part.

To be sure we are not doing a greedy grab, we use `(?=[a-zA-Z0-9])`, which is a [lookbehind](https://stackoverflow.com/questions/2973436/regex-lookahead-lookbehind-and-atomic-groups). That means we check if the next character is `[a-zA-Z0-9]` as it should, but it's not included in the string that's being deleted.

With the `for` loop the result is:

```
INBOX
INBOX.Trash
INBOX.Spam
INBOX.Drafts
```

I'm not sure I can explain this in a short comment. Would `string formatting with lookbehind` be better?